scripts/propagate-i18n-tags.js

```diff
@@ -0,0 +1,151 @@
+const localeFiles = fs
+    .readdirSync(
+        path.join(__dirname, '../app/locale')
+    )
+    .filter((fileName) => IGNORE_FILES.indexOf(fileName) === -1);
+
+for (const localeFile of localeFiles) {
+    console.log(`Processing ${localeFile}`);
+
+    // Load raw source file
+    const localeSource = fs.readFileSync(
+        path.join(__dirname, `../app/locale/${localeFile}`),
+        { encoding: 'utf8'}
+    );
+
+    // Load locale
+    const localeContents = require(`../app/locale/${localeFile}`);
+    const localeTranslations = localeContents.translation || {};
+
+    // Initialie rewrite of file with first line
+    const rewrittenLines = [
+        `module.exports = nativeDescription: "${localeContents.nativeDescription}", englishDescription: ` +
+            `"${localeContents.englishDescription}", translation:`
+    ];
+
+    // For each category within the locale
+    for (const enCategoryName of Object.keys(enTranslations)) {
+        const enCategory = enTranslations[enCategoryName];
+        const catIsPresent = (typeof localeTranslations[enCategoryName] !== 'undefined');
+        const localeCategory = localeTranslations[enCategoryName] || {};
+
+        rewrittenLines.push('');
+
+        // Add the category line, commenting it out if it does not exist in the locale file
+        const categoryCommentPrefix = (!catIsPresent)  ? '#' : '';
+        rewrittenLines.push(`${categoryCommentPrefix}  ${enCategoryName}:`);
+
+        // For each tag within the category
+        for (const enTagName of Object.keys(enCategory)) {
+            const localeTranslation = localeCategory[enTagName];
+            const tagIsPresent = (typeof localeTranslation !== 'undefined');
+
+            // Prepare the comment for the tag if it exists.  Note that this will propagate {change} tag from en locale
+            let comment = '';
+            if (comments[enCategoryName] && comments[enCategoryName][enTagName]) {
+                comment = comments[enCategoryName][enTagName];
+            }
+
+            const commentedTagRegex = new RegExp(`#\\s+${escapeRegexp(enTagName)}:`);
+            if (localeSource.search(commentedTagRegex) >= 0) {
+                // If the translation is commented out in the locale fine, make sure it is not marked as changed.  A
+                // translation is not marked as changed until it is uncommented in a locale file.  Once it is
+                // uncommented in a translation file, the translation is considered active and changes should be
+                // tracked
+                comment = comment.replace(CHANGE_PATTERN, '');
+            } else {
+                const tagIsMarkedChangeRegex = new RegExp(`^\\s+"?${escapeRegexp(enTagName)}"?:\\s".*"\\s*` +
+                    `#\\s*${escapeRegexp(CHANGE_MARKER)}\\s*$`, 'm');
+
+                // If locale file has tag marked as change and comment is not already marked change,
+                // add change to comment
```

Comment:
We can probably add more context to this comment, it looks a bit confusing.
Something like: `if en file has tag marked as change and this locale file doesnt, then add change comment to local file`?

Reply:
Updated